discovery/gossiper.go

```diff
@@ -1032,6 +1032,7 @@ func (d *AuthenticatedGossiper) networkHandler() {
 
 			// Once a new block arrives, we updates our running
 			// track of the height of the chain tip.
+			d.Lock()
```

Comment:
Given that we need to synchronize `d.bestHeight` under the lock as well, it seems like we can stop using it atomically now?

Reply:
Indeed I believe we can. I didn't do it originally to keep the diff small but I can amend this PR if you think it's appropriate.

Reply:
Go for it.